### PR TITLE
update status.go (fix errcheck issue)

### PR DESCRIPTION
**Pull request by TheInvincibleRalph**
#### Ticket Link

Fixes https://github.com/mattermost/mattermost/issues/29099

```release-note
NONE
```

### CODE REVIEW

**Review by hanzei — CHANGES_REQUESTED** [review]
There is one case that needs checking:
````
Error: channels/app/status.go:101:22: Error return value of `user.SetCustomStatus` is not checked (errcheck)
	user.SetCustomStatus(cs)
```

Can you please take a look?

**Comment by TheInvincibleRalph** [comment]
> There is one case that needs checking:
> 
> ```
> Error: channels/app/status.go:101:22: Error return value of `user.SetCustomStatus` is not checked (errcheck)
> 	user.SetCustomStatus(cs)
> ```
> 
> Can you please take a look?
> ```

Fixed. Thank you!

File: server/channels/app/status.go [hunk 1]
```diff
@@ -98,7 +98,8 @@ func (a *App) SetCustomStatus(c request.CTX, userID string, cs *model.CustomStat
 		return err
 	}
 
-	user.SetCustomStatus(cs)
+	if err = user.SetCustomStatus(cs); err != nil {
+		c.Logger().Error("Failed to set custom status", mlog.String("userID", userID), mlog.Err(err))
```
Comment (hanzei) [inline server/channels/app/status.go:102]: The code doesn't compile:

```
Error: channels/app/status.go:115:36: syntax error: unexpected request in argument list; possibly missing comma or )
Error: channels/app/status.go:116:12: syntax error: unexpected := in composite literal; possibly missing comma or }
```

**Review by hanzei — CHANGES_REQUESTED** [review]

File: server/channels/app/status.go (hunk 1 above)
Comment (TheInvincibleRalph) [reply server/channels/app/status.go:102]: image

The problem is from here. I don't see anything wrong with the argument list yet it flags a missing comma.

Comment (hanzei) [reply server/channels/app/status.go:102]: You should take a look at the code that I highlighted. The error is there.

Comment (TheInvincibleRalph) [reply server/channels/app/status.go]: Oh. Seen. The function was expecting a closing curly brace. My apologies.

**Review by hanzei — APPROVED** [review]
Nice one :+1:

**Review by hanzei — APPROVED** [review]
Nice one :+1:

**Review by harshilsharma63 — APPROVED** [review]

**Pull request merged**